edward/util.py

```diff
@@ -18,10 +28,20 @@ def cumprod(xs):
     return result
 
 def dot(x, y):
-    """
-    x is M x N matrix and y is N-vector, or
-    x is M-vector and y is M x N matrix
-    """
+    """Compute dot product between a Tensor matrix and a Tensor vector.
+
+    Parameters
+    ----------
+    x : tf.Tensor
+        `M x N` matrix or `M` vector (respectively)
+    y : tf.Tensor
+        `M` vector or `M x N` matrix (respectively)
```

Comment:
This is a bit confusing to understand. May have to be more verbose in describing the correspondence of dimensions.